cmake/modules/Findndctl.cmake

```diff
@@ -0,0 +1,41 @@
+# - Find libndctl
+# Find the ndctl libraries and includes
+#
+# ndctl_INCLUDE_DIR - where to find libndctl.h etc.
+# ndctl_LIBRARIES - List of libraries when using ndctl.
+# ndctl_FOUND - True if ndctl found.
+
+
+find_path(ndctl_INCLUDE_DIR ndctl/libndctl.h)
+
+if(ndctl_INCLUDE_DIR AND EXISTS "${ndctl_INCLUDE_DIR}/libndctl.h")
+  foreach(ver "MAJOR" "MINOR" "RELEASE")
+    file(STRINGS "${ndctl_INCLUDE_DIR}/libndctl.h" ndctl_VER_${ver}_LINE
+      REGEX "^#define[ \t]+ndctl_VERSION_${ver}[ \t]+[0-9]+[ \t]+.*$")
+    string(REGEX REPLACE "^#define[ \t]+ndctl_VERSION_${ver}[ \t]+([0-9]+)[ \t]+.*$"
+      "\\1" ndctl_VERSION_${ver} "${ndctl_VER_${ver}_LINE}")
+    unset(${ndctl_VER_${ver}_LINE})
+  endforeach()
+  set(ndctl_VERSION_STRING
+    "${ndctl_VERSION_MAJOR}.${ndctl_VERSION_MINOR}.${ndctl_VERSION_RELEASE}")
+endif()
+
+find_library(ndctl_LIBRARY NAMES ndctl libndctl)
```

Comment:
Yes,  I have read it. My understanding is to list all possible names, such as `ndctl(first choise) ` +` libndctl.` But `libndctl `exists in my system. why it failed when I just add `libndctl`? Is it something I missed? This question is just because I am curious and want to learn cmake.  If you can solve my doubts, thank you very much.